regimes/pt/tags.go

```diff
@@ -0,0 +1,312 @@
+package pt
+
+import (
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/i18n"
+	"github.com/invopop/gobl/tax"
+)
+
+// Tax exemption tags
+const (
+	TagExempt cbc.Key = "exempt"
+	TagM01    cbc.Key = "m01"
+	TagM02    cbc.Key = "m02"
+	TagM04    cbc.Key = "m04"
+	TagM05    cbc.Key = "m05"
+	TagM06    cbc.Key = "m06"
+	TagM07    cbc.Key = "m07"
+	TagM09    cbc.Key = "m09"
+	TagM10    cbc.Key = "m10"
+	TagM11    cbc.Key = "m11"
+	TagM12    cbc.Key = "m12"
+	TagM13    cbc.Key = "m13"
+	TagM14    cbc.Key = "m14"
+	TagM15    cbc.Key = "m15"
+	TagM16    cbc.Key = "m16"
+	TagM19    cbc.Key = "m19"
+	TagM20    cbc.Key = "m20"
+	TagM21    cbc.Key = "m21"
+	TagM25    cbc.Key = "m25"
+	TagM30    cbc.Key = "m30"
+	TagM31    cbc.Key = "m31"
+	TagM32    cbc.Key = "m32"
+	TagM33    cbc.Key = "m33"
+	TagM40    cbc.Key = "m40"
+	TagM41    cbc.Key = "m41"
+	TagM42    cbc.Key = "m42"
+	TagM43    cbc.Key = "m43"
+	TagM99    cbc.Key = "m99"
```

Comment:
I named the tags directly after the TaxExemptionCode they represent in Portugal. I've noticed in other countries, we use more meaningful names. In this case, though, the name for this tags isn't as obvious (at least to me) than in the other countries. Let me know if you have any suggestion for naming these tags differently

Reply:
For Italy, I used ChatGPT to come up with short names for complex tax categories, then tweaked them manually. I did wonder, though, whether using the actual codes as keys would have been clearer, and whether inventing our own internal key actually increased conceptual bloat without adding too much value. 

I guess it depends on the purpose of these keys and who will be using them primarily. Under what circumstances would a user or a developer wish the keys were given more meaningful names? 

Curious to hear what @samlown thinks about this.

Reply:
Exactly my thoughts while working on it. Putting myself on the shoes of a GOBL developer, I don't know if I would rather use a code that is well-known in my domain (e.g. `exempt+m01`) or some other code which only exists in the GOBL world but that may potentially be more meaningful (e.g. `exempt+outlay`). For illustration, see how our own users refer to the M99 exemption explicit by its code [here](https://invopop.slack.com/archives/C04FYMD6SLE/p1679386203186589?thread_ts=1679066241.190119&cid=C04FYMD6SLE))

Now, I've been working on this a bit more and came up with a couple of alternatives to name these tags:

Código | Tag (Legal Reference) | Tag (Meaningful)
-- | -- | --
M01 | exempt+civa+article16 | exempt+outlay
M02 | exempt+dl198-90 | exempt+intrastate-export
M04 | exempt+civa+article13 | exempt+imports
M05 | exempt+civa+article14 | exempt+exports
M06 | exempt+civa+article15 | exempt+suspension-scheme
M07 | exempt+civa+article9 | exempt+internal-operations
M09 | exempt+civa+article62b | exempt+small-retail-scheme
M10 | exempt+civa+article57 | exempt+exempt-scheme
M11 | exempt+dl346-85 | exempt+tobacco-scheme
M12 | exempt+dl221-85 | exempt+margin-scheme+travel
M13 | exempt+dl199-66+second-hand | exempt+margin-scheme+second-hand
M14 | exempt+dl199-66+art | exempt+margin-scheme+art
M15 | exempt+dl199-66+antiques | exempt+margin-scheme+antiques
M16 | exempt+riti+article14 | exempt+goods-transmission
M19 | exempt+other | exempt+other
M20 | exempt+civa+article59 | exempt+flat-rate-scheme
M21 | exempt+civa+article72 | exempt+non-deductible
M25 | exempt+civa+article38 | exempt+consignment-goods
M30 | exempt+civa+article2i | exempt+reverse-charge+waste
M31 | exempt+civa+article2j | exempt+reverse-charge+civil-eng
M32 | exempt+civa+article2l | exempt+reverse-charge+greenhouse
M33 | exempt+civa+article2m | exempt+reverse-charge+woods
M40 | exempt+civa+article6 | exempt+reverse-charge+b2b
M41 | exempt+riti+article8 | exempt+reverse-charge+intraeu
M42 | exempt+dl27-2007 | exempt+reverse-charge+real-estate
M43 | exempt+dl362-99 | exempt+reverse-charge+gold
M99 | exempt+non-taxable | exempt+non-taxable

I tend to like the last column better, because it hints you about its meaning (although, it's less precise than the legal reference or the exemption code). What do you think?

Reply:
Yea, I think you're on to something here. I'm starting to lean on the side of preferring local codes as keys and relying on `Name` to provide clarity and meaning. 

This feels like a pretty important conversation though. We should set a precedence for the future and definitely adopt a consistent approach across regimes.

Reply:
I'd go with the last one. However, my intuition here is the the `M99` value is the equivalent of not defining anything. Basically, its the value I'd use in the output whenever nothing else is defined, as opposed to being an option of itself. Would that work?

Reply:
Regarding Luca's comment here. Perhaps we need to clarify the intent of the longer keys. From one side you do have the "meaning" which can be transferred in a good key, but maybe thats not the only use-case, there is also the factor of error checking and typos. Its much easier to skim a document and mix up `M12` with `M21`, something which wouldn't happen with `exempt+margin-scheme+travel` and `exempt+non-deductible`.

So I would argue that its not too much of a problem to loose some meaning in the key, provided its easy to differentiate between the options available.

Reply:
@samlown both `M99` and `M19` seem indeed to be catch-all categories to be used for exemptions not covered by the rest of the options but still legal. In this case, though, I tend lean towards explicitness over implicitness. Firstly, because there are two possible defaults (M99 and M19) and secondly, because the omission of an exemption tag in the GOBL may not be intentional. If we simply default to M99 when no VAT and no other exemption is given, then we're increasing the chances of integrators not realising they need to provide the right exemption reason. That's why I'll keep M99 on the list, and won't select it by default.

Reply:
Thanks for clarifying Sam, that makes sense.

Reply:
I don't think `M19` is a catch-all. My take is that it's a specific reason as to why the line is exempt from VAT, whereas `M99` applies when something specific is happening and there is no VAT, which is slightly different.

What I'm suggesting as a possibility is that `M99` could be applied automatically when the VAT tax combo is not there, assuming its a required field in Portugal. Happy with either approach though.